src/Commands/User.php

```diff
@@ -0,0 +1,693 @@
+                    break;
+
+                case 'list':
+                    $this->list($username, $email);
+                    break;
+
+                case 'addgroup':
+                    $this->addgroup($group, $username, $email);
+                    break;
+
+                case 'removegroup':
+                    $this->removegroup($group, $username, $email);
+                    break;
+            }
+        } catch (RuntimeException $e) {
+            return EXIT_ERROR;
+        }
+
+        return EXIT_SUCCESS;
+    }
+
+    private function setTables(): void
+    {
+        /** @var Auth $config */
+        $config       = config('Auth');
+        $this->tables = $config->tables;
+    }
+
+    private function setValidationRules(): void
+    {
+        $validationRules = new RegistrationValidationRules();
+
+        $rules = $validationRules->get();
+
+        $this->validationRules = [
+            'username' => $rules['username'],
+            'email'    => $rules['email'],
+            'password' => $rules['password'],
+        ];
+    }
+
+    /**
+     * Asks the user for input.
+     *
+     * @param string       $field      Output "field" question
+     * @param array|string $options    String to a default value, array to a list of options (the first option will be the default value)
+     * @param array|string $validation Validation rules
+     *
+     * @return string The user input
+     */
+    private function prompt(string $field, $options = null, $validation = null): string
+    {
+        return self::$io->prompt($field, $options, $validation);
+    }
+
+    /**
+     * Outputs a string to the cli on its own line.
+     */
+    private function write(
+        string $text = '',
+        ?string $foreground = null,
+        ?string $background = null
+    ): void {
+        self::$io->write($text, $foreground, $background);
+    }
+
+    /**
+     * Create a new user
+     *
+     * @param string|null $username User name to create (optional)
+     * @param string|null $email    User email to create (optional)
+     */
+    private function create(?string $username = null, ?string $email = null): void
+    {
+        $data = [];
+
+        if ($username === null) {
+            $username = $this->prompt('Username', null, $this->validationRules['username']);
+        }
+        $data['username'] = $username;
+
+        if ($email === null) {
+            $email = $this->prompt('Email', null, $this->validationRules['email']);
+        }
+        $data['email'] = $email;
+
+        $password        = $this->prompt('Password', null, $this->validationRules['password']);
+        $passwordConfirm = $this->prompt(
+            'Password confirmation',
+            null,
+            $this->validationRules['password']
+        );
+
+        if ($password !== $passwordConfirm) {
+            $this->write("The passwords don't match", 'red');
+
+            throw new RuntimeException("The passwords don't match");
```

Comment:
Not good. Do not remove the user from the process. Display new form to re-enter data.